pkg/wasm/service.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package wasm
+
+import (
+	"math"
+	"os"
+	"path/filepath"
+	"sync"
+	"sync/atomic"
+)
+
+import (
+	"github.com/mitchellh/mapstructure"
+
+	"mosn.io/proxy-wasm-go-host/common"
+
+	"mosn.io/proxy-wasm-go-host/proxywasm"
+
+	"mosn.io/proxy-wasm-go-host/wasmer"
+)
+
+import (
+	"github.com/apache/dubbo-go-pixiu/pkg/context/http"
+	"github.com/apache/dubbo-go-pixiu/pkg/model"
+)
+
+// WasmService manages a collection of WasmInstances of the same type.
+type WasmService struct {
+	rootContextID      int32
+	contextIDGenerator int32
+	name               string
+	path               string
+	mutex              sync.Mutex
+	model              model.WasmService
+	instancePool       sync.Pool
+}
+
+type WasmConfig struct {
+	Path string `yaml:"path" json:"path,omitempty"`
+}
+
+type ABIContextWrapper struct {
```

Comment:
Some comments for this struct? I found some methods are also not commented on.

Reply:
Do private methods also need to be commented on?